lib/pure/os.nim

```diff
@@ -666,6 +589,82 @@ proc isAbsolute*(path: string): bool {.rtl, noSideEffect, extern: "nos$1".} =
   elif defined(posix):
     result = path[0] == '/'
 
+proc parentDir*(path: string): string {.
+  noSideEffect, rtl, extern: "nos$1".} =
+  ## Returns the parent directory of `path`.
+  ##
+  ## Trailing ``.`` and ``/`` are resolved before taking the parent dir, see
+  ## examples below for corner cases.
+  ##
+  ## The remainder can be obtained with `lastPathPart(path) proc
+  ## <#lastPathPart,string>`_.
+  ##
+  ## See also:
+  ## * `relativePath proc <#relativePath,string,string>`_
+  ## * `splitPath proc <#splitPath,string>`_
+  ## * `tailDir proc <#tailDir,string>`_
+  ## * `parentDirs iterator <#parentDirs.i,string>`_
+  runnableExamples:
+    assert parentDir("") == ""
+    when defined(posix):
+      assert parentDir("/usr/local/bin") == "/usr/local"
+      assert parentDir("foo//bar//.") == "foo"
+      assert parentDir("foo") == "."
+      assert parentDir("/foo") == "/"
+      assert parentDir(".") == ""
+      assert parentDir("/") == ""
+  if path == "": return ""
+  result = path
+  while true:
+    normalizePathEnd(result)
+    let (dir, name, ext) = splitFile(result)
+    if name == ".." or (name.len == 0 and ext.len == 0): return ""
+    if name == ".":
+      result = dir
+      continue
+    if dir == "" and name != "":
+      return "."
```

Comment:
Really? What is this "." directory and if it is one why is ".." not a directory?